src/test/ui/lint/suggestions.stderr

```diff
@@ -22,7 +22,7 @@ warning: variable does not need to be mutable
   --> $DIR/suggestions.rs:36:13
    |
 36 |         let mut a = (1); // should suggest no `mut`, no parens
-   |             ---^^
+   |             ----^
```

Comment:
I did something similar in https://github.com/rust-lang/rust/pull/45741

The terminal output should not be showing whitespace.

Reply:
This is the opposite problem to that. The span needs to point to the whitespace as well, so that the suggestion removes that whitespace. The suggestion content does not have trailing whitespaces.

Reply:
Right. But the human readable output should not highlight the whitespace. Only json should have it